spring-cloud-kubernetes-client-autoconfig/src/main/java/org/springframework/cloud/kubernetes/client/KubernetesClientAutoConfiguration.java

```diff
@@ -42,9 +42,9 @@ public class KubernetesClientAutoConfiguration {
 
 	@Bean
 	@ConditionalOnMissingBean
-	public ApiClient apiClient(KubernetesClientProperties properties) {
```

Comment:
@ryanjbaxter I am in the blur. May be I spent too much time on this and my vision is somehow biased, but I don't understand what is going on. If we inject `KubernetesClientProperties` in here, properties are not injected. 

You left somehow the same comment in `KubernetesInformerDiscoveryConfiguration`:

```// Injecting KubernetesDiscoveryProperties here would cause it to be
			// initialize too early
			// Instead get the all-namespaces property value from the Environment directly
			boolean allNamespaces = environment.getProperty("spring.cloud.kubernetes.discovery.all-namespaces",
					Boolean.class, false);
```

but for the love of God, I do not understand why that happens.

Reply:
We can move this to main, since it moves properties to a record and then l'll try to figure out what is going on. Either way, this never actually worked the way we thought.

Reply:
Yeah, I've run into this several times.  Its all about the order in which things get created.  If something gets created too early before the environment is populated then the resulting configuration properties bean won't be initialized properly.  By using the environment we are basically getting the property "on demand".  I am curious though do you know what change is causing the initialization so early?

Reply:
Here is the thing: no change at all. This was never working. I debugged in main branch, and properties are initialized with all "defaults". So when we had a default constructor, that was called. Now that we don't ( because of record ), things break. 

I'm a bit busy with my _actual_ work this week :),  but I'll try to figure out end to end what the problem is

Reply:
Got it can you create an issue for it and if I have some cycles I can try and take a look as well.  Appreciate all your help!